news/get_text.py

```diff
@@ -0,0 +1,125 @@
+    help="Path to output directory where raw pages are downloaded.",
+)
+parser.add_argument(
+    "--output_dir",
+    default="data/news-propublica/",
+    help="Path to output directory for processed data.",
+)
+parser.add_argument(
+    "--version",
+    type=int,
+    default=1,
+    help="Version of the subset",
+)
+parser.add_argument(
+    "--index_path",
+    default=None,
+    help="File that list of all pages",
+)
+parser.add_argument(
+    "--overwrite",
+    action="store_true",
+    help="Should we overwrite previously downloaded copies?",
+)
+parser.add_argument(
+    "--filename", default="pro.jsonl.gz", help="The base filename for our data."
+)
+parser.add_argument(
+    "--shard_size", type=int, default=1, help="Size, in GB, for each shard."
+)
+parser.add_argument(
+    "--tag", type=str, default="div", help="Tag for the article or content"
+)
+parser.add_argument(
+    "--attrs", type=json.loads, default=None, help="dict of attributes"
+)
+parser.add_argument(
+    "--num_workers",
+    default=None,
+    help="Number of workers",
+)
+
+def get_record(page_index, input_dir=None, date=None, tag="div", attrs=None):
+    idx = page_index["idx"]
+    url = page_index["url"]
+    filename = page_index["filename"]
+
+    html_path = os.path.join(input_dir, filename)
+    if os.path.exists(html_path):
+        page_text = utils.get_text_from_page(html_path=html_path, tag=tag, attrs=attrs)
+
+        return {
+            "id": idx,
+            "text": page_text,
+            "source": url,
+            "added": date,
+            "metadata": {
+                "license": "Creative Commons License (CC BY-NC-ND 3.0)",
+            }
+        }
+    else:
+        return None
+
+def main(args):
+
+    input_dir = os.path.join(args.input_dir)
+    Path(input_dir).mkdir(parents=True, exist_ok=True)
+    output_dir = os.path.join(args.output_dir, f"v{args.version}")
+    Path(output_dir).mkdir(parents=True, exist_ok=True)
+
+    current_datetime = datetime.now()
+    date = f"{current_datetime.year}-{current_datetime.month}-{current_datetime.day}"
+
+    if args.index_path:
+        with jsonlines.open(args.index_path) as reader:
+            page_index = [line for line in reader]
+    else:
+        pagelist_path = os.path.join(input_dir, "pagelist.jsonl")
+        if os.path.isfile(pagelist_path) and args.overwrite is False:
+            with jsonlines.open(pagelist_path) as reader:
+                page_index = [line for line in reader]
+    
+    # TODO Save HTML files
+    # Then process/extract
+    get_record_fn = partial(get_record, input_dir=input_dir, date=date, tag=args.tag, attrs=args.attrs)
+    num_workers = mp.cpu_count() if args.num_workers is None else args.num_workers
+    if num_workers == 1:
+        page_data = list(map(get_record_fn, tqdm(page_index)))
+    else:
+        with mp.Pool(num_workers) as p:
+            page_data = []
+            for data in p.map(get_record_fn, tqdm(page_index)):
```

Comment:
I think this can be replaced with `list(p.map(get_record_fn, tqdm(page_index))` instead of manually looping through the results.

Also I don't think you even need to have the `list` call, just `page_data = p.map(...)` as you aren't using `imap`

Reply:
Thanks. Adjusted.